File: Packages/StewardMDWatchCore/Sources/StewardMDWatchCore/Engines/RateCoach.swift
import Foundation

/// Maps an estimated compression rate to the AHA target band (design §3.1). Copy is
/// strictly rate-matching guidance — never a CPR-quality/adequacy verdict (§0 safety).
public enum RateCoach {
    public static let lower = 100
    public static let upper = 120

    public static func zone(forRateCPM rate: Int, active: Bool) -> RateZone {
        guard active, rate > 0 else { return .idle }
        if rate < lower { return .tooSlow }
        if rate > upper { return .tooFast }
        return .onTarget
    }

    public static func guidance(_ z: RateZone) -> String {
        switch z {
        case .tooSlow: return "Faster"
        case .onTarget: return "On target"
        case .tooFast: return "Slower"
        case .idle: return "—"
        }
    }
}

File: android/wear/src/main/kotlin/in/stewardmd/wear/codeblue/RateCoach.kt
package `in`.stewardmd.wear.codeblue

/**
 * Maps an estimated compression RATE to the AHA target band. Port of WatchCore RateCoach. Copy is
 * strictly rate-matching guidance — NEVER a CPR-quality/adequacy verdict, NEVER depth (design §0).
 */
enum class RateZone { Idle, TooSlow, OnTarget, TooFast }

object RateCoach {
    const val LOWER = 100
    const val UPPER = 120

    fun zone(rateCpm: Int, active: Boolean): RateZone {
        if (!active || rateCpm <= 0) return RateZone.Idle
        return when {
            rateCpm < LOWER -> RateZone.TooSlow
            rateCpm > UPPER -> RateZone.TooFast
            else -> RateZone.OnTarget
        }
    }

    /** Rate-matching guidance only. */
    fun label(zone: RateZone): String = when (zone) {
        RateZone.Idle -> "—"
        RateZone.TooSlow -> "Push faster"
        RateZone.OnTarget -> "On target"
        RateZone.TooFast -> "Push slower"
    }
}
